plugins/modules/gitlab_project.py

```diff
@@ -471,7 +514,14 @@ def update_project(self, project, arguments):
         for arg_key, arg_value in arguments.items():
             if arguments[arg_key] is not None:
                 if getattr(project, arg_key) != arguments[arg_key]:
-                    setattr(project, arg_key, arguments[arg_key])
+                    if (arg_key == 'container_expiration_policy'):
+                        old_val = getattr(project, arg_key)
+                        old_val.pop('next_run_at', None)
```

Comment:
This is modifying the attribute of that object. I would avoid that.

Also that attribute might have more values in the future that `container_expiration_policy` does not have, at which point the check below will always fail. It's probably better to only explicitly compare values that are available in `container_expiration_policy`, and only set these (and not the whole of `container_expiration_policy`).